pkg/install/install.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2019 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package install
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/heptio/velero/pkg/client"
+)
+
+// kindToResource translates a Kind (mixed case, singular) to a Resource (lowercase, plural) string.
+// This is to accomodate the dynamic client's need for an APIResource, as the Unstructured objects do not have easy helpers for this information.
+var kindToResource = map[string]string{
+	"CustomResourceDefinition": "customresourcedefinitions",
+	"Namespace":                "namespaces",
+	"ClusterRoleBinding":       "clusterrolebindings",
+	"ServiceAccount":           "serviceaccounts",
+	"Deployment":               "deployments",
+	"DaemonSet":                "daemonsets",
+	"BackupStorageLocation":    "backupstoragelocations",
+	"VolumeSnapshotLocation":   "volumesnapshotlocations",
+}
+
+// Install creates resources on the Kubernetes cluster.
+func Install(factory client.DynamicFactory, resources *unstructured.UnstructuredList, logger *logrus.Logger) error {
+	for _, r := range resources.Items {
+		// TODO: do we want to use the logger, or just a print?
```

Comment:
Switched this to an `io.Writer` so that `pkg/install` could be used as a library, too. I think this strikes a decent balance for our use case and what others might use it for.